apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/post-published-modal/index.tsx

```diff
@@ -17,29 +17,30 @@ const PostPublishedModal: React.FC = () => {
 		select( 'core/editor' ).isCurrentPostPublished()
 	);
 	const previousIsCurrentPostPublished = useRef( isCurrentPostPublished );
-	const siteHasNeverPublishedPost = useSelect( ( select ) =>
-		select( 'automattic/wpcom-welcome-guide' ).getSiteHasNeverPublishedPost()
+	const shouldShowFirstPostPublishedModal = useSelect( ( select ) =>
+		select( 'automattic/wpcom-welcome-guide' ).getShouldShowFirstPostPublishedModal()
 	);
 	const [ isOpen, setIsOpen ] = useState( false );
 	const closeModal = () => setIsOpen( false );
-	const { fetchSiteHasNeverPublishedPost, setSiteHasNeverPublishedPost } = useDispatch(
-		'automattic/wpcom-welcome-guide'
-	);
+	const {
+		fetchShouldShowFirstPostPublishedModal,
+		setShouldShowFirstPostPublishedModal,
+	} = useDispatch( 'automattic/wpcom-welcome-guide' );
 
 	useEffect( () => {
-		fetchSiteHasNeverPublishedPost();
-	}, [ fetchSiteHasNeverPublishedPost ] );
+		fetchShouldShowFirstPostPublishedModal();
+	}, [ fetchShouldShowFirstPostPublishedModal ] );
 	useEffect( () => {
 		// If the user never published any post before and the current post status changed to publish,
```

Comment:
should probably update the comment too
```suggestion
		// If the user is set to see the first post modal and current post status changes to publish,
```